### ISSUE TITLE
Inconsistent tab size, impossible to align

### ISSUE THREAD

**Issue by jekyllgrim** [typora/typora-issues#3451]
I'm confused there are no reports I could find on this, but here it is: tabs in Typora don't seem to do fulfill their primary function—keeping text aligned. This is how it looks:
![image](https://user-images.githubusercontent.com/42670463/80360681-5e76df80-8888-11ea-97f6-a770911a1529.png)
It's pretty clear that line 2 is indented differently from lines 1 and 3.
Here's a worse example:
![image](https://user-images.githubusercontent.com/42670463/80360830-9a11a980-8888-11ea-9d13-e4bbfe56e851.png)
These are just impossible to align.

I don't understand how and why this is happening, I've never seen that in a text editor. Moreover, when uploaded to github, these tabs go completely mad and end up all over the place. (The initial tabs at the beginning of lines, however, look correct after upload.)

**Comment by abnerlee** [comment]
covered in #579, relates #498

Currently tab are 2~4 spaces based on your setting. you can remove one more whitespace in line 2 in pic 1 and add 2 more space in line 2 and 1 more space in line 3 for indent.

**Comment by jekyllgrim** [comment]
I'm sorry, I genuinely don't understand your answer. However many tabs I add or delete, it can't align those lines properly. You aren't suggesting to add regular _spaces_ to fix indentation, are you? Because that's not a solution; the whole point of tabs is to align the text without the use of spaces. #579 seems to be related but it's closed without a solution.

**Comment by vicmarto** [comment]
tab is clearly inconsistent, even in 2025 (version 1.12.1 (7516)).
:(
